test/unit/associations/belongs-to-many.test.js

```diff
@@ -110,6 +113,230 @@ describe(Support.getTestDialectTeaser('belongsToMany'), function() {
     });
   });
 
+  describe('pseudo associations', function () {
+    it('should setup belongsTo relations to source and target from join model with defined foreign/other keys', function () {
+      var Product = this.sequelize.define('Product', {
```

Comment:
It only affects `identifierField` which is really want to move to later in the query building process - so i skipped it ;p